lib/spack/spack/relocate.py

```diff
@@ -775,6 +775,9 @@ def relocate_text(files, prefixes, concurrency=32):
 
     args = []
     for filename in files:
+        # Passing a directory to _replace_prefix_text will still throw error
+        if os.path.isdir(filename):
+            continue
```

Comment:
```suggestion
```

Reply:
How can you just remove this? That won't fix the bug.

Reply:
```suggestion
        # Passing a directory to _replace_prefix_text will still throw error
        if os.path.isdir(filename) and os.path.islink(filename):
            continue
```

Reply:
Thanks @trws !